### DISCUSSION TITLE
Different layouts for base/admin part of the app?

### DISCUSSION THREAD

**Discussion by djrobby** [jeremyevans/roda#298] [General]
Given the following app structure:

```
app/
├─ views/
│  ├─ admin/
│  │  ├─ _layouts/
│  │  │  ├─ admin.slim
│  │  ├─ index.slim
│  ├─ base/
│  │  ├─ _layouts/
│  │  │  ├─ base.slim
│  │  ├─ index.slim
config.ru
app.rb
```

How can I utilize the `render` plugin to use a different `layout` based on the routing tree branch? Here's a simple `app.rb` example of what I'm looking to accomplish:

```rb
# app.rb
require "roda"

class App < Roda
  plugin :render, engine: :slim, views: 'app/views', layout: '_layouts/base'
  
  hash_branch 'admin' do |r|
    plugin :render, engine: :slim, views: 'app/views/admin', layout: '_layouts/admin'
    r.hash_routes(:admin)
  end
  
  route do |r|
    r.root do
      view 'base/index'
    end
    
    r.hash_routes(:root)
    r.hash_branches
  end
end
```

However, it seems that I can't override a plugin's settings inside another `hash_branch`. I do know that I can override the layout per route but I'd rather have some kind of a global switch that applies that correct layout based on the branch. Any help would be greatly appreciated! Thank you in advance 🙏

**Comment by djrobby** [comment]
Nevermind, I found what I needed. Looks like all I needed to do was use the `view_options` plugin as following:
```ruby
require "roda"

class App < Roda
  plugin :render, engine: :slim, views: 'app/views', layout: '_layouts/base', default_encoding: 'UTF-8'
  plugin :view_options
  
  hash_branch 'admin' do |r|
    set_view_subdir 'admin'
    set_layout_options(template: 'admin/_layouts/admin')
    r.hash_routes(:admin)
  end
  
  route do |r|
    r.root do
      view 'base/index'
    end
    
    r.hash_routes(:root)
    r.hash_branches
  end
end
```
So awesome!!! Love Roda 💕

**Comment by jeremyevans** [comment]
You could also try the `hash_branch_view_subdir` plugin, which will automatically handle the `set_view_subdir 'admin'` part.
